File: programmers/고득점Kit/해시/의상.js
// 의상
// https://school.programmers.co.kr/learn/courses/30/lessons/42578

function solution(clothes) {
  let answer = 1;
  let types = {};
  
  clothes.forEach((c) => types[c[1]] = (types[c[1]] || 0) + 1);

  for (const t in types) {
    answer *= types[t] + 1;
  }

  return answer - 1 ;
}

console.log(solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]]))
console.log(solution([["crow_mask", "face"], ["blue_sunglasses", "face"], ["smoky_makeup", "face"]]))

File: programmers/Lv.2/의상.py
# 의상
# https://school.programmers.co.kr/learn/courses/30/lessons/42578
from itertools import combinations

def solution(clothes):
    answer = 1
    type = {}
    for c in clothes:
        if c[1] in type:
            type[c[1]] += 1
        else:
            type[c[1]] = 1
    
    for value in type.values():
        answer *= value + 1

    return answer - 1

print(solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]]))
print(solution([["crow_mask", "face"], ["blue_sunglasses", "face"], ["smoky_makeup", "face"]]))